docs/connections.md

```diff
@@ -6,9 +6,9 @@ You can connect to MongoDB with the `mongoose.connect()` method.
 mongoose.connect('mongodb://127.0.0.1:27017/myapp');
 ```
 
-This is the minimum needed to connect the `myapp` database running locally
-on the default port (27017). If connecting fails on your machine, try using
-`127.0.0.1` instead of `localhost`.
+This is the minimum needed to connect the `myapp` database running locally on the default port (27017).
+For local MongoDB databases, we recommend using `127.0.0.1` instead of `localhost`.
+That is because Node.js 18 and up prefer IPv6 addresses, which means, on many machines, Node.js will resolve `localhost` to the IPv6 address `::1` and Mongoose will be unable to connect.
```

Comment:
```suggestion
That is because Node.js 18 and up prefer IPv6 addresses, which means, on many machines, Node.js will resolve `localhost` to the IPv6 address `::1` and Mongoose will be unable to connect, unless the mongodb instance is running with ipv6 enabled.
```
maybe note that this is only the case if the instance is not running with ipv6 enabled